### ISSUE TITLE
thread 'main' panicked at 'Mismatch between definition and access of `pretty`. Could not downcast to TypeId

### ISSUE THREAD

**Issue by lmammino** [lmammino/jwtinfo#32]
On linux. Specifically:

```
cat /etc/os-release 
NAME="CentOS Stream"
VERSION="8"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="8"
PLATFORM_ID="platform:el8"
PRETTY_NAME="CentOS Stream 8"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:8"
HOME_URL="https://centos.org/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux 8"
REDHAT_SUPPORT_PRODUCT_VERSION="CentOS Stream"
```

## Full error

```
RUST_BACKTRACE=1 jwtinfo $TOKEN
thread 'main' panicked at 'Mismatch between definition and access of `pretty`. Could not downcast to TypeId { t: 324675245860759320943513204350442872190 }, need to downcast to TypeId { t: 7428646492878894209665195255548636123 }
', /home/luciano.mammino/.cargo/registry/src/index.crates.io-6f17d22bba15001f/clap_builder-4.4.5/src/parser/error.rs:32:9
stack backtrace:
   0: rust_begin_unwind
             at /rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3/library/std/src/panicking.rs:593:5
   1: core::panicking::panic_fmt
             at /rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3/library/core/src/panicking.rs:67:14
   2: clap_builder::parser::matches::arg_matches::ArgMatches::get_flag
   3: jwtinfo::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

**Comment by WhiskeyTuesday** [comment]
Same crash here

![image](https://github.com/lmammino/jwtinfo/assets/3824702/5e0fc683-c29e-42cd-bff2-dca39acf855e)

**Comment by lmammino** [comment]
Hey @WhiskeyTuesday, thanks for reporting this. I suspect this is a `clap` issue. Shortly releasing a new `0.4.1` version using an updated version of `clap`. Let's see if that fixes the problem... 🤞🏽

**Comment by lmammino** [comment]
Tested this with:

```bash
docker run -it centos:8.4.2105 bash
# inside the container
sed -i 's/mirrorlist/#mirrorlist/g' /etc/yum.repos.d/CentOS-*
sed -i 's|#baseurl=http://mirror.centos.org|baseurl=http://vault.centos.org|g' /etc/yum.repos.d/CentOS-*
yum update -y
yum groupinstall -y 'Development Tools'
curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
export PATH=$PATH:$HOME/.cargo/bin
cargo install jwtinfo
jwtinfo --version # jwtinfo 0.4.1
jwtinfo eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c
```

Still seeing:

> Mismatch between definition and access of `pretty`. Could not downcast to TypeId { t: 324675245860759320943513204350442872190 }, need to downcast to TypeId { t: 7428646492878894209665195255548636123 }

This probably requires some changes at the clap config level then 😢

**Comment by lmammino** [comment]
After reverting to the old `clap` builder pattern approach this seems to be now fixed (`v0.4.2`). Closing it now. Feel free to re-open if the problem persists!
